dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -0,0 +1,317 @@
+package com.dotcms.vanityurl.business;
+
+import com.dotcms.contenttype.model.type.BaseContentType;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.model.type.VanityUrlContentType;
+import com.dotcms.services.VanityUrlServices;
+import com.dotcms.util.VanityUrlUtil;
+import com.dotcms.vanityurl.model.CachedVanityUrl;
+import com.dotcms.vanityurl.model.DefaultVanityUrl;
+import com.dotcms.vanityurl.model.VanityUrl;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.collect.ImmutableList;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+/**
+ * Implementation class for the {@link VanityUrlAPI}.
+ *
+ * @author oswaldogallango
+ * @version 4.2.0
+ * @since June 12, 2017
+ */
+public class VanityUrlAPIImpl implements VanityUrlAPI {
+
+    private final ContentletAPI contentletAPI = APILocator.getContentletAPI();
+    private final VanityUrl cache404VanityUrl = new DefaultVanityUrl();
+
+    private static final String GET_VANITY_URL_BASE_TYPE =
+            "+baseType:" + BaseContentType.VANITY_URL.getType();
+    private static final String GET_ALL_VANITY_URL = GET_VANITY_URL_BASE_TYPE + " +working:true";
+    private static final String GET_ACTIVE_VANITY_URL =
+            GET_VANITY_URL_BASE_TYPE + " +live:true +deleted:false";
+    private static final String GET_VANITY_URL_BY_URI = GET_VANITY_URL_BASE_TYPE + " +conHost:";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "GET_VANITY_URL_BY_URI" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)